### DISCUSSION TITLE
how do i get endpoint like flask  by request.endpoint

### DISCUSSION THREAD

**Discussion by un-human** [fastapi/fastapi#6733] [Questions] [answered]
### question

- fastapi, how do i get endpoint; like flask  by request.endpoint ??

**Comment by ycd** [comment]
There is a similar request object.  
```python
async def dummy(request: Request):
    return request.url.path
```

**Comment by un-human** [comment]
> There is a similar request object.
> 
> ```python
> async def dummy(request: Request):
>     return request.url.path
> ```

request.url.path Not what I wanted；
i wanted the router_name and the function_name , like flask

**Comment by ycd** [comment]
Hmm you mean something like this

```python
from starlette.routing import Match

routes = request.app.router.routes
for route in routes:
    match, scope = route.matches(request)
    if match == Match.FULL:
        path = route.path
```

**Comment by un-human** [comment, marked as answer]
> Hmm you mean something like this
> 
> ```python
> from starlette.routing import Match
> 
> routes = request.app.router.routes
> for route in routes:
>     match, scope = route.matches(request)
>     if match == Match.FULL:
>         path = route.path
> ```

```python
from fastapi import APIRouter, FastAPI
app = FastAPI()
router = APIRouter(prefix='/user')

@router.get("/route/{id}")
async def route_time(id: int,request: Request):
    # here,i want get the  "user" of router   and “route_time” of the function_name , not "/route/{id}" of url_path
    pass

```

**Comment by ycd** [comment]
I assume it worked.

**Comment by tiangolo** [comment]
Thanks for the help here @ycd ! 👏  🙇 

Thanks for coming back and closing the issue @stringk245  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
